BlueDBOnDisk/src/main/java/io/bluedb/disk/file/BlueObjectInput.java

```diff
@@ -0,0 +1,64 @@
+package io.bluedb.disk.file;
+
+import java.io.Closeable;
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import io.bluedb.api.exceptions.BlueDbException;
+import io.bluedb.disk.serialization.BlueSerializer;
+
+public class BlueObjectInput<T> implements Closeable {
+
+	private final Path path;
+	private final BlueSerializer serializer;
+	private final DataInputStream dataInputStream;
+			
+	public BlueObjectInput(BlueReadLock<Path> readLock, BlueSerializer serializer) throws BlueDbException {
+		this.path = readLock.getKey();
+		this.serializer = serializer;
+		dataInputStream = openDataInputStream(path.toFile());
+	}
+
+	public T next() throws EOFException, BlueDbException {
+		int objectLength;
+		try {
+			objectLength = dataInputStream.readInt();
+			byte[] buffer = new byte[objectLength];
```

Comment:
It would be nice if we can figure out a good way to re-use our buffer

Reply:
There's probably a way but I don't want to spend a lot of time on it until we determine if we're for sure going to go this route